File: Backend/App/models/appointment.py
from sqlalchemy import Column,String,Integer,Boolean,ForeignKey,DateTime
from Backend.App.core.database import Base
from Backend.App.models.person import Person

class Appointment(Base):
    __tablename__ ="Appointment"

    AppointmentId = Column(Integer,primary_key=True,index=True)
    DoctorId=Column(Integer,ForeignKey(Person.PersonId))
    PatientId=Column(Integer,ForeignKey(Person.PersonId))
    Time_of_Appointment = Column(DateTime)
    Is_Closed = Column(Boolean)

File: DotNetBackend/ViveksHomoeoClinic/ViveksHomoeoClinic/Models/EFModel/Appointment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ViveksHomoeoClinic.Models.EFModel
{
    public class Appointment
    {
        [Key]
        public int AppointmentId { get; set; }
        [Required]
        public int DoctorId { get; set; }
        [Required]
        public int PatientId { get; set; }

        [ForeignKey(nameof(DoctorId))]
        public Staff Doctor { get; set; }

        [ForeignKey(nameof(PatientId))]
        public Patient Patient { get; set; }
        public DateTime AppointmentTime { get; set; }
        public bool IsClosed { get; set; }
    }
}
